.circleci/config.yml

```diff
@@ -21,21 +63,95 @@ jobs:
           name: "Run Tests for CLI"
           command: cd cli && go test
       - run:
-          name: "Build CLI"
-          command: cd cli && go build -o yurl
+          name: "Run Tests for webapp"
+          command: cd webapp && go test
+      - save_cache:
+          key: go-mod-v1
+          paths:
+            - "/go/pkg/mod"
+  build-binaries-cli:
+    parameters:
+      publish:
+        type: boolean
+        default: false
+    docker:
+      - image: cimg/go:1.15
+    steps:
+      - checkout
       - run:
-          name: "Check Version for CLI"
-          command: cd cli && ./yurl version
+          name: "Install GoReleaser"
+          command: |
+            curl -sSL "https://github.com/goreleaser/goreleaser/releases/download/v0.141.0/goreleaser_Linux_x86_64.tar.gz" | sudo tar -xz -C /usr/local/bin goreleaser
+            goreleaser --version
+      - unless:
+          condition: << parameters.publish >>
+          steps:
+            - run:
+                name: "Build Binaries with GoReleaser"
+                command: goreleaser --snapshot --skip-publish --rm-dist
+      - when:
+          condition: << parameters.publish >>
+          steps:
+            - run:
+                name: "Build Binaries & Publish with GoReleaser"
+                command: goreleaser
+      - persist_to_workspace:
+          root: "."
+          paths:
+            - "dist"
+      - run:
+          name: "Display CLI version"
+          command: dist/yurl_linux_amd64/yurl version
+  build-snap:
+    parameters:
+      publish-edge:
+        type: boolean
+        default: false
+      publish-stable:
+        type: boolean
+        default: false
+    docker:
+      - image: cibuilds/snapcraft:core18
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          name: "Build Snaps for Each Architecture"
+          command: |
+            # Generate a version file that Snapcraft can use
+            dist/yurl_linux_amd64/yurl version --short > VERSION
+            snapcraft
+      - when:
+          condition: << parameters.publish-edge >>
+          steps:
+            - run:
+                name: "Publish to Snap Store in the Edge Channel"
+                command: |
+                  # Snapcraft Tokens are valid for 1 year. This one expires August 10th, 2021
+                  echo $SNAPCRAFT_LOGIN_FILE | base64 --decode --ignore-garbage | snapcraft login --with -
```

Comment:
Did you setup `$SNAPCRAFT_LOGIN_FILE`?

Reply:
No I haven't, I haven't been able to generate a snap yet. Still looking into this.